packages/babel-helper-create-class-features-plugin/src/misc.js

```diff
@@ -84,9 +82,14 @@ export function injectInitialization(path, constructor, nodes, renamer) {
 
 export function extractComputedKeys(ref, path, computedPaths, file) {
   const declarations = [];
-
   for (const computedPath of computedPaths) {
-    computedPath.traverse(classFieldDefinitionEvaluationTDZVisitor, {
+    if (computedPath.get("key").isReferencedIdentifier()) {
+      handleClassTDZ(computedPath.get("key"), {
+        classBinding: path.node.id && path.scope.getBinding(path.node.id.name),
+        file,
+      });
+    }
+    computedPath.get("key").traverse(classFieldDefinitionEvaluationTDZVisitor, {
       classBinding: path.node.id && path.scope.getBinding(path.node.id.name),
       file,
     });
```

Comment:
Instead of rewriting the object, you can extract it to a variable.

Reply:
Done. PTAL.